synapse/lib/autodoc.py

```diff
@@ -323,17 +327,35 @@ def docStormTypes(page, docinfo, linkprefix, islib=False, lvl=1,
 
             if isinstance(rtype, dict):
                 rname = rtype.get('type')
-                assert rname == 'function', f'Unknown type: loclname={loclname} rname={rname}'
+
+                if isinstance(rname, dict):
+                    raise AssertionError(f'rname as dict not supported loclname={loclname} rname={rname}')
+
+                isstor = False
+                isfunc = False
+                isgtor = False
```

Comment:
Is `isgtor` used anywhere?

Reply:
Currently it is not. Some of this is future proofing for more robust generation.